common/repo/GitRepo.java

```diff
@@ -365,23 +302,47 @@ public void removeBranch(String branch) {
   @Override
   public String getFileContents(String commitId, String path) {
     CommandResult commandResult = runCommand("show " + commitId + ":" + path);
-    String result = commandResult.stdout;
-    int lastIndexOfNewLineSymbol = result.lastIndexOf("\n");
-    if (lastIndexOfNewLineSymbol >= 0) {
-      result =
-          new StringBuilder(result)
-              .replace(lastIndexOfNewLineSymbol, lastIndexOfNewLineSymbol + 1, "")
-              .toString();
-    }
-    return result;
+    return commandResult.stdout;
+  }
+
+  @Override
+  public String currentBranch() {
+    return runCommand("rev-parse --abbrev-ref HEAD").stdout.trim();
   }
 
   public void init() {
     runCommand("init");
   }
 
-  public String currentBranch() {
-    return runCommand("rev-parse --abbrev-ref HEAD").stdout.trim();
+  public void addFile(String path) {
+    runCommand("add " + path);
+  }
+
+  public String getHeadCommitId() {
+    CommandResult commandResult = runCommand("rev-parse HEAD");
+    return commandResult.stdout.replace("\n", "");
+  }
+
+  private void switchToMasterBranch() {
+    if (!currentBranch().equals("master")) {
+      switchBranch("master");
+    }
+  }
+
+  @VisibleForTesting
+  public ImmutableList<String> getTagList() {
+    CommandResult commandResult = runCommand("tag");
+    try {
+      return splitLines(commandResult.stdout);
+    } catch (IllegalStateException e) {
```

Comment:
The stack trace will include `getTagList()`, so you don't have to explicitly catch & rethrow, plus you're removing the original exception.